db-service/lib/cqn2sql.js

```diff
@@ -832,9 +832,9 @@ class CQN2SQLRenderer {
       case 'number': return `${val}` // REVISIT for HANA
       case 'object':
         if (val === null) return 'NULL'
-        if (val instanceof Date) return `'${val.toISOString()}'`
-        if (val instanceof Readable); // go on with default below
-        else if (Buffer.isBuffer(val)); // go on with default below
+        if (val instanceof Date) val = val.toJSON() // returns null if invalid
+        else if (val instanceof Readable); // go on with default below
+        else if (Buffer.isBuffer(val)) val = val.toString('base64')
```

Comment:
Why did that work before and not any longer?

Reply:
Good catch was a side effect from a merge conflict.